streams/src/main/java/org/apache/kafka/streams/state/internals/OffsetCheckpoint.java

```diff
@@ -90,6 +94,7 @@ public void write(final Map<TopicPartition, Long> offsets) throws IOException {
                 fileOutputStream.getFD().sync();
             }
 
+            LOG.trace("Swapping tmp checkpoint file {} {}", temp.toPath(), file.toPath());
```

Comment:
Having these logs would have demystified a large part of the prior (misdirected) investigation, since we were never sure whether the tmp file existed or not, or what was going on.